app/main.py

```diff
@@ -2,24 +2,68 @@
 
 
 class IntegerRange:
-    pass
+    """Descriptor data validation"""
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{value} must be in range:"       # noqa
+                             f" {self.min_amount} - {self.max_amount}")
+        setattr(instance, self.protected_name, value)
```

Comment:
This is a potential code quality issue. In the `__set__` method of your descriptor, you are not checking the type of the value being set. This could lead to unexpected behavior or errors if a non-integer value is used. Consider adding a check to ensure that the value is indeed an integer before proceeding with the range validation.